lms/djangoapps/courseware/masquerade.py

```diff
@@ -172,32 +172,23 @@ def is_masquerading_as_specific_student(user, course_key):  # pylint: disable=in
     return bool(course_masquerade and course_masquerade.user_name)
 
 
-def get_masquerading_group_info(user, course_key):
-    """
-    If the user is masquerading as belonging to a group, then this method returns
-    two values: the id of the group, and the id of the user partition that the group
-    belongs to. If the user is not masquerading as a group, then None is returned.
-    """
-    course_masquerade = get_course_masquerade(user, course_key)
-    if not course_masquerade:
-        return None, None
-    return course_masquerade.group_id, course_masquerade.user_partition_id
-
-
 def get_masquerading_user_group(course_key, user, user_partition):
     """
-    If the current user is masquerading as a generic student in a specific group, return that group
+    If the current user is masquerading as a generic learner in a specific group, return that group.
+    If the user is not masquerading as a group, then None is returned.
     """
-    group_id, user_partition_id = get_masquerading_group_info(user, course_key)
-    if user_partition_id == user_partition.id and group_id is not None:
-        try:
-            return user_partition.get_group(group_id)
-        except NoSuchUserPartitionGroupError:
-            return None
-    # The user is masquerading as a generic student. We can't show any particular group.
+    course_masquerade = get_course_masquerade(user, course_key)
+    if course_masquerade:
+        if course_masquerade.user_partition_id == user_partition.id and course_masquerade.group_id is not None:
```

Comment:
Thanks, this actually is much easier to understand! I always found the previous "if user_partition_id == user_partition.id" confusing.